### PR TITLE
Fix and add tests for my cooperation details and cooperation details #1742

### PR DESCRIPTION
**Pull request by OliaHrytsak**
This tests failed time to time, because of speed of internet connection - the tests were synchronous, while rendering components is asynchronous (Axios request)

- Add async/await to tests
- Add few tests to raise coverague

### CODE REVIEW

File: tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx [hunk 1]
```diff
@@ -59,6 +59,16 @@ const cooperationMock = {
   updatedAt: '2024-01-12T11:28:34.397Z'
 }
 
+vi.mock(
+  '~/containers/my-cooperations/cooperation-notes/CooperationNotes',
+  () => ({
+    __esModule: true,
```
Comment (iva-stasia) [inline tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx:65]: Please note that we don't need to use `__esModule: true` in the current Vitest version.

File: tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx [hunk 2]
```diff
@@ -70,42 +80,53 @@ describe('CooperationDetails', () => {
...
+    const tab1 = await screen.findByText('cooperationsPage.tabs.activities')
 
     expect(tab1).toBeInTheDocument()
 
-    const tab2 = screen.getByText('cooperationsPage.tabs.details')
-
-    fireEvent.click(tab2)
+    const tab2 = await screen.findByText('cooperationsPage.tabs.details')
 
+    await act(async () => {
```
Comment (iva-stasia) [inline tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx:105]: Please, could you explain why do we need to use `act` here?
The same for all the following tests.

Comment (Tolik170) [reply tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx:105]: It is good practice to use act when the state can update, but I don`t understand why you make a callback inside to by async

File: tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx (hunk 1 above)
Comment (OliaHrytsak) [reply tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx]: Thank you! I've removed it

File: tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx (hunk 2 above)
Comment (OliaHrytsak) [reply tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx]: Thank for you advice. I dived diper in this topic and made some changes

**Review by iva-stasia — APPROVED** [review]

**Review by t-valentyna — APPROVED** [review]

**Review by GeorgeShvab — APPROVED** [review]

**Review by byavd — APPROVED** [review]

**Pull request merged**